symbolic-symcache/src/writer.rs

```diff
@@ -318,19 +317,22 @@ where
         Ok(())
     }
 
-    /// Adds a function to this SymCache.
+    /// Cleans up a function by recursively removing all empty inlinees, then inserts it into
+    /// the writer. Does nothing if the function is empty itself.
     ///
     /// Functions **must** be added in ascending order using this method. This emits a function
     /// record for this function and for each inlinee recursively.
-    pub fn add_function(&mut self, mut function: Function<'_>) -> Result<(), SymCacheError> {
+    pub fn clean_and_insert_function(
```

Comment:
Can we leave this `add_function`? The fact that the function is cleaned should be considered an implementation detail.

_nit_: Same comment about the subject line applies.